src/main/java/fi/fmi/avi/model/AviationCodeListUser.java

```diff
@@ -1,433 +1,501 @@
+        BECOMING(1),
+        TEMPORARY_FLUCTUATIONS(2),
+        FROM(3),
+        PROBABILITY_30(4),
+        PROBABILITY_30_TEMPORARY_FLUCTUATIONS(5),
+        PROBABILITY_40(6),
+        PROBABILITY_40_TEMPORARY_FLUCTUATIONS(7);
+
+        public static TAFChangeIndicator fromInt(final int code) {
+            switch (code) {
+                case 1: return BECOMING;
+                case 2: return TEMPORARY_FLUCTUATIONS;
+                case 3: return FROM;
+                case 4: return PROBABILITY_30;
+                case 5: return PROBABILITY_30_TEMPORARY_FLUCTUATIONS;
+                case 6: return PROBABILITY_40;
+                case 7: return PROBABILITY_40_TEMPORARY_FLUCTUATIONS;
+                default: throw new IllegalArgumentException("No value for code " + code);
+            }
+        }
+
+        private final int code;
+
+        TAFChangeIndicator(final int code) {
+            this.code = code;
+        }
+
+        public int getCode() {
+            return this.code;
+        }
+    }
+
+    enum ColorState {
+        BLU(2500, 8000),
+        WHT(1500, 5000),
+        GRN(700, 3700),
+        YLO1(500, 2500),
+        YLO2(300, 1600),
+        AMB(200, 800),
+        RED(200, 800, BELOW),
+        BLACK(-1, -1, null, true),
+        BLACKBLU(2500, 8000, null, true),
+        BLACKWHT(1500, 5000, null, true),
+        BLACKGRN(700, 3700, null, true),
+        BLACKYLO1(500, 2500, null, true),
+        BLACKYLO2(300, 1600, null, true),
+        BLACKAMB(200, 800, null, true),
+        BLACKRED(200, 800, BELOW, true);
+
+        private final double minCloudHeight;
+        private final double minVisibility;
+        private final boolean unusable;
+        private final RelationalOperator operator;
+
+        ColorState(final double cloudHeight, final double visibility) {
+            this(cloudHeight, visibility, null, false);
+        }
+
+        ColorState(final double cloudHeight, final double visibility, final RelationalOperator operator) {
+            this(cloudHeight, visibility, operator, false);
+        }
+
+        ColorState(final double cloudHeight, final double visibility, final RelationalOperator operator, final boolean unusable) {
+            this.minCloudHeight = cloudHeight;
+            this.minVisibility = visibility;
+            this.operator = operator;
+            this.unusable = unusable;
+        }
+
+        public double getMinCloudHeight() {
+            return minCloudHeight;
+        }
+
+        public double getMinVisibility() {
+            return minVisibility;
+        }
+
+        public boolean isUnusable() {
+            return unusable;
+        }
+
+        public RelationalOperator getOperator() {
+            return operator;
+        }
+    }
+    
+    enum PermissibleUsage {OPERATIONAL, NON_OPERATIONAL}
+    
+    enum PermissibleUsageReason {TEST, EXERCISE}
+
+    //From: http://codes.wmo.int/49-2/SigWxPhenomena
+    enum AeronauticalSignificantWeatherPhenomenon {

```

Comment:
It would be better if the enum names would not be abbreviated, only the text codes